KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java

```diff
@@ -470,15 +488,38 @@ public void onEvent(IKalturaEvent event){
 			{
 				case STREAM_PUBLISHED:
 					streamEvent = (KalturaStreamEvent) event;
+					logger.info("Calling KalturaLiveManager onPublish");
 					onPublish(streamEvent.getEntryId(), streamEvent.getServerIndex(), streamEvent.getApplicationName());
 					break;
+				case STREAM_READY_FOR_PLAYBACK:
+					streamEvent = (KalturaStreamEvent) event;
+					logger.info("Calling KalturaLiveManager onStreamReadyForPlayback");
+					onStreamReadyForPlayback(streamEvent.getEntryId(), streamEvent.getServerIndex());
 
+					//
 				default:
 					break;
 			}
 		}
 	}
 
+	protected  void onStreamReadyForPlayback(String entryId, final KalturaMediaServerIndex serverIndex) {
+		logger.debug("onStreamReadyForPlayback [" + entryId + "]");
+
+		LiveEntryCache liveEntryCache = null;
+
+		synchronized (entries) {
+
+			if (entries.containsKey(entryId)) {
+				liveEntryCache = entries.get(entryId);
+				liveEntryCache.index = serverIndex;
```

Comment:
Yes we do.
We're calling liveEntryCache.setReadyForPlayback(), and in this method we're calling setEntryMediaServer which is where we send isLive. setEntryMediaServer requires serverIndex as one of its parameters therefore we're saving it as a private instance in the calss liveEntryCache